apps/contribution/contracts/Contribution.sol

```diff
@@ -4,21 +4,29 @@ import "@aragon/os/contracts/apps/AragonApp.sol";
 import "@aragon/os/contracts/kernel/IKernel.sol";
 
 interface IToken {
-  function mintFor(address contributorAccount, uint256 amount, uint256 contributionId) public;
+  function mintFor(address contributorAccount, uint256 amount, uint32 contributionId) public;
+}
+
+interface ContributorInterface {
+  function getContributorAddressById(uint32 contributorId) public view returns (address);
+  function getContributorIdByAddress(address contributorAccount) public view returns (uint32);
+  // TODO Maybe use for validation
+  // function exists(uint32 contributorId) public view returns (bool);
 }
 
 contract Contribution is AragonApp {
   bytes32 public constant ADD_CONTRIBUTION_ROLE = keccak256("ADD_CONTRIBUTION_ROLE");
   bytes32 public constant VETO_CONTRIBUTION_ROLE = keccak256("VETO_CONTRIBUTION_ROLE");
 
   bytes32 public constant KERNEL_APP_ADDR_NAMESPACE = 0xd6f028ca0e8edb4a8c9757ca4fdccab25fa1e0317da1188108f7d2dee14902fb;
+
   // ensure alphabetic order
   enum Apps { Contribution, Contributor, Proposal, Token }
   bytes32[4] public appIds;
 
   struct ContributionData {
-    address contributor;
-    uint256 amount;
+    uint32 contributorId;
+    uint32 amount;
```

Comment:
The `mintFor` takes a `uint256`

Reply:
Yes, but we're converting it before minting in `claim()`. Nobody has to mine more than 2.2 billion kredits at once, so the locally stored amount can be a smaller format.